include/mxnet/lib_api.h

```diff
@@ -277,6 +283,14 @@ struct MXTensor {
     return size;
   }
 
+  /*! \brief helper function to compare two MXTensors */
+  inline bool isSame(const MXTensor &oth) {
```

Comment:
should we override operator==? since we won't support C anyway

Reply:
done

Reply:
I think operator== is confusing. For a tensor object, == usually means value comparison.
In the future, we may add other operators !=, <, >, etc.
It may be better and more consistent with MXNet NDArray API to use ‘isSame’.

Reply:
comparing object is how c++ is doing for vector and usually for struct, and in NDArray we don't have operators !=, <, > either, so I don't think it is going to be confusing

Reply:
I think I agree with @wkcn here, == should compare the values of the tensor not the "state" of the tensor (data_ptr, versionID, etc)

@mseth10 @eric-haibin-lin @haojin2 what do you guys think?

Reply:
For C++ vector container, `operator==` compares the values.
```c++
#include <iostream>
#include <vector>
using namespace std;

int main() {
  vector<int> a{1,2,3};
  vector<int> b{1,2,3};
  vector<int> c{1,2,4};
  cout << (a == b) << endl; // 1
  cout << (a == c) << endl; // 0
  return 0;
}
```